awscrt/auth.py

```diff
@@ -255,6 +255,32 @@ def new_chain(cls, providers):
         binding = _awscrt.credentials_provider_new_chain(providers)
         return cls(binding)
 
+    @classmethod
+    def new_python(cls, py_provider):
+        """
+        Creates a provider that sources credentials from a python class.
+
+        The python provider needs to have a method :get_credential(), which returns
+        the credential as a dictionary like (Expiration and SessionToken are optional):
+        {
+            "AccessKeyId": "accesskey",
+            "SecretAccessKey": "secretAccessKey",
+            "SessionToken": "....",
+            "Expiration": 1247169778 // timepoint, in seconds since epoch
+        }
```

Comment:
how will this work with something like ECDSA keys? @bretambrose how you feel about this one?